<html><head><meta charset="utf-8"></meta></head><body><h1 class="article-title" style="text-align:center;">Python3断言</h1><div style="width:100%;float:left;" class="article-content">   
 <div> 
  <div>
    断言是一种理智检查，当程序的测试完成，你可以打开或关闭。 
  </div> 
 </div> 
 <p style="text-align:justify;"> 断言的最简单的方法就是把它比作&nbsp;raise-if&nbsp;语句&nbsp;(或者更准确，加 raise-if-not&nbsp;声明).&nbsp;一个表达式进行测试，如果结果出现&nbsp;false，将引发异常。 </p> 
 <div>
   断言是由&nbsp;assert&nbsp;语句，在Python中新的关键字，在Python1.5版本中引入使用的关键字。 
 </div> 
 <div>
   程序员常常放置断言来检查输入的有效，或在一个函数调用后检查有效的输出。 
 </div> 
 <h3> assert&nbsp;语句 </h3> 
 <p style="text-align:justify;"> 当它遇到一个断言语句，Python评估计算之后的表达式，希望是&nbsp;true&nbsp;值。如果表达式为 false，Python 触发 AssertionError 异常。 </p> 
 <div>
   断言的语法是&nbsp;- 
 </div> 
 <pre>assert Expression[, Arguments]&nbsp;</pre> 
 <p style="text-align:justify;"> 如果断言失败，Python使用&nbsp;ArgumentExpression&nbsp;作为AssertionError异常的参数。AssertionError异常可以被捕获，并用 try-except语句处理类似其他异常，但是，如果没有处理它们将终止该程序并产生一个回溯。 </p> 
 <h3> 示例 </h3> 
 <p style="text-align:justify;"> 这里是一个把从开氏度到华氏度的温度转换函数。 </p>   
 <pre>#!/usr/bin/python3
def KelvinToFahrenheit(Temperature):
    assert (Temperature &gt;= 0),"Colder than absolute zero!"
    return ((Temperature-273)*1.8)+32

print (KelvinToFahrenheit(273))
print (int(KelvinToFahrenheit(505.78)))
print (KelvinToFahrenheit(-5))</pre> 
 <div>
   当执行上面的代码，它产生以下结果&nbsp;- 
 </div> 
 <pre>32.0
451
Traceback (most recent call last):
File "test.py", line 9, in 
print KelvinToFahrenheit(-5)
File "test.py", line 4, in KelvinToFahrenheit
assert (Temperature &gt;= 0),"Colder than absolute zero!"
AssertionError: Colder than absolute zero!</pre> 
 <div> 
  <br> 
 </div>
 <br>      
</div></body></html>